src/utils/codapPhone/index.ts

```diff
@@ -156,28 +157,88 @@ export function getDataFromContext(
   );
 }
 
-function createBareDataset(
-  name: string,
-  attrs: CodapAttribute[]
+// export async function getAllCasesFromContext(contextName: string) {
+//   const context = await getDataContext(contextName);
+//   return new Promise<unknown>((resolve, reject) => phone.call);
+// }
+
+function getDataContext(contextName: string): Promise<DataContext> {
+  return new Promise<DataContext>((resolve, reject) =>
+    phone.call(
+      {
+        action: CodapActions.Get,
+        resource: resourceFromContext(contextName),
+      },
+      (response: GetContextResponse) => {
+        if (response.success) {
+          resolve(response.values);
+        } else {
+          reject(new Error(`Failed to get context ${contextName}`));
+        }
+      }
+    )
+  );
+}
+
+// In the returned collections, parents show up as numeric ids, so before
+// reusing, we need to look up the names of the parent collections.
+function normalizeParentNames(collections: ReturnedCollection[]): Collection[] {
+  const normalized = [];
+  for (const c of collections) {
+    let newParent;
+    if (c.parent) {
+      newParent = collections.find(
+        (collection) => collection.id === c.parent
+      )?.name;
+    }
+
+    // Only copy the non-internal fields from the collection's attributes
+    // In particular, exclude `cid` from attributes
+    const attrs = c.attrs?.map((attr) => {
+      return {
+        name: attr.name,
+        title: attr.title,
+        type: attr.type,
+        colormap: attr.colormap,
+        description: attr.description,
+        editable: attr.editable,
+        formula: attr.formula,
+        hidden: attr.hidden,
+        precision: attr.precision,
+        unit: attr.unit,
+      };
+    });
```

Comment:
@stardust66 @P-bibs if either of you have a better way of doing this feel free to change--I am trying to make sure only the fields outlined in the `BaseAttribute` type actually get copied into the new attributes.

The objects in `c.attrs` given to us by CODAP have more fields that we want to avoid copying.